### ISSUE TITLE
Props spreading causes a false positive type error

### ISSUE THREAD

**Issue by 1898182** [microsoft/TypeScript#51950]
# Bug Report

### 🔎 Search Terms

`spread`  `props destructuring`  `...props`

### 🕗 Version & Regression Information

`4.4.2`  `4.9.4` and `Nightly`

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `props`

### ⏯ Playground Link

[Codesandbox link](https://codesandbox.io/s/infallible-oskar-46mqmz?file=/src/App.tsx:0-469)

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.0.0-dev.20221218#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5ICwAUAJYC2ADnPgC6gBKSAhjAwGb5wWgDkhNnT6lRJOpCoooAYR5UACtyrYAvKADepUDtg4yAGwSEAdgC4mrdgDpmQgHKIkAblIBfVyVLwTmBrPlQdQAKDT1DYyQTABpQa3iqZWw3CwDqJTgVAEoggD5NbV1COgBXfBNQAB4qTXjrRMzMN1yNGH0jUzdK4Cpczw8xHz9QAEEqGpCc1XytEl1QYrKK4ML5nUq0mr9IAyRVDTD4A1oLACJCBFPQNzdYAxZMTHsWCj3TikgAWnhqb-vH065VZrebVXIAZR4KB8dCidG6vWBoOAmyBc10WX6QA)

### 💻 Code

```tsx
type MyCompProps = {
    children: React.ReactNode;
};

const MyComp = ({ children, ...props }: MyCompProps) => {
    return <p {...props}>{children}</p>;
};

const App = () => {
    return (
        <MyComp style={{ color: "red" }} className="my-comp-class">
            <p>Some content</p>
        </MyComp>
    );
}
```

### 🙁 Actual behavior

When rendering this code, everything is working just fine in the Browser. The paragraph will be in red text color and will have the class attribute with the value `my-comp-class`. However, TypeScript will underline in red the `style` attribute and complain about its type.

```
Type '{ children: Element; style: { color: string; }; className: string; }' is not assignable to type 'IntrinsicAttributes & MyCompProps'.
  Property 'style' does not exist on type 'IntrinsicAttributes & MyCompProps'.
```

### 🙂 Expected behavior

TypeScript doesn't complain.
